### PR TITLE
fixed Twig deprecation notices

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

To avoid deprecation notices when upgrading to Twig 1.23.

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php [hunk 1]
```diff
@@ -29,12 +29,18 @@ class TwigRendererEngine extends AbstractRendererEngine implements TwigRendererE
      */
     private $template;
 
+    public function __construct(array $defaultThemes = array(), \Twig_Environment $environment)
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php]: This is where we have the BC break as the environment is now required.

Comment (fabpot) [inline src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php]: Which breaks Silex for instance.

**Comment by fabpot** [comment]
I can see two other options:

 * Deprecate `initRuntime` in Twig 2.1 but that means waiting for Twig 3.0 to implement the performance optimization :(

 * Adding a new interface in Twig 1.x for the `initRuntime` method that needs to be implemented to avoid the deprecation notice. We would then deprecate this new interface in Twig 2.1.

**Comment by nicolas-grekas** [comment]
I don't know if it's relevant here (I read quickly), but we didn't try to avoid deprecation notices in 2.3.

File: src/Symfony/Bridge/Twig/Form/TwigRendererEngineInterface.php [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
 /**
  * @author Bernhard Schussek <bschussek@gmail.com>
+ *
+ * @deprecated Deprecated in 2.8, to be removed in 3.0.
```
Comment (Tobion) [inline src/Symfony/Bridge/Twig/Form/TwigRendererEngineInterface.php]: There should not be new deprecatations in a maintanance branch

**Comment by fabpot** [comment]
Totally new implements where I just implement the new interface from twig/twig#1886. Another PR for 2.8 will do the other changes.

File: src/Symfony/Bridge/Twig/Extension/FormExtension.php [hunk 3]
```diff
@@ -21,7 +21,7 @@
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
-class FormExtension extends \Twig_Extension
+class FormExtension extends \Twig_Extension implements \Twig_Extension_InitRuntimeInterface
```
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/Extension/FormExtension.php]: Where is this interface defined? Travis fails when trying to load it.

Comment (xabbuh) [inline src/Symfony/Bridge/Twig/Extension/FormExtension.php]: Ah sorry, I missed the link to twigphp/Twig#1886.

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

Can we merge here to make Travis builds pass again?

**Comment by fabpot** [comment]
I need to release Twig 1.23 first.

File: src/Symfony/Bundle/SecurityBundle/composer.json [hunk 4]
```diff
@@ -34,7 +34,7 @@
         "symfony/process": "~2.0,>=2.0.5",
         "symfony/validator": "~2.2",
         "symfony/yaml": "~2.0,>=2.0.5",
-        "twig/twig": "~1.20|~2.0"
+        "twig/twig": "~1.23|~2.0"
```
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/composer.json]: Is this really needed? The `LogoutUrlExtension` does not make use of the `setEnvironment()` method.

**Comment by Tobion** [comment]
@fabpot  @nicolas-grekas This produces a deprecation warning through the DebugClassLoader on every request in symfony standard edition.

> The Symfony\Bridge\Twig\Extension\FormExtension class implements Twig_Extension_InitRuntimeInterface that is deprecated to be removed in 3.0

**Comment by stof** [comment]
``Twig_Extension_InitRuntimeInterface`` should not be marked as deprecated in Twig IMO, as it is precisely meant to be implemented by extensions not wanting the deprecation warning

**Comment by xabbuh** [comment]
This happens when using the dev version of Twig 2.0 I guess (the interface is not tagged as deprecated in Twig 1.23, is it?).

**Pull request merged**